### PR TITLE
networkctl: introduce --runtime for editing network config under /run/

### PR DESCRIPTION
**Pull request by YHNdnzj**

### CODE REVIEW

**Review by yuwata — APPROVED** [review]

**Comment by YHNdnzj** [comment]
Added a new commit and fixed the test. PTAL.

File: test/units/testsuite-74.networkctl.sh [hunk 1]
```diff
@@ -80,6 +85,9 @@ networkctl cat @test2:network | cmp - <(networkctl cat "$NETWORK_NAME")
 EDITOR='cp' script -ec 'networkctl edit @test2 --drop-in test2.conf' /dev/null
 cmp "+4" "/etc/systemd/network/${NETWORK_NAME}.d/test2.conf"
 
+sleep 1
```
Comment (YHNdnzj) [inline test/units/testsuite-74.networkctl.sh:88]: So without the `sleep` here, the test seems to fail consistently. @yuwata Is there any chance that the reloading of networkd is not synchronous?

Comment (yuwata) [reply test/units/testsuite-74.networkctl.sh:88]: Unfortunately, reloading is not synchronous.
https://github.com/systemd/systemd/blob/97f9a25cfd87b82feec1e67d2ffdf2be1aaa349e/test/test-network/systemd-networkd-tests.py#L738-L743

Comment (yuwata) [reply test/units/testsuite-74.networkctl.sh:88]: I think this should be fixed in a future release.

Comment (yuwata) [reply test/units/testsuite-74.networkctl.sh:88]: -> #31247.

**Pull request merged**